app/javascript/utils/api.js

```diff
@@ -0,0 +1,11 @@
+const API = window.API;
+
+export const createSubnet = (values, emsId, routerRef) => API.post(`/api/providers/${emsId}/cloud_subnets`, {
+  action: 'create',
+  resource: {...values, router_ref: routerRef},
+}).then(response => {
+  let res = response['results'][0];
+  window.add_flash(res.message, res.success ? 'success' : 'error');
```

Comment:
I think it is safer to cycle through all result messages so you don't miss anything. `response.forEach(item => window.add_flash...)`.